vumi/persist/txriak_manager.py

```diff
@@ -155,31 +159,60 @@ def set_user_metadata(self, usermeta):
         self._riak_obj.usermeta = usermeta
 
     def get_bucket(self):
-        return VumiTxRiakBucket(self._riak_obj.bucket)
+        return VumiTxRiakBucket(self._riak_obj.bucket, self._manager)
 
     # Methods that touch the network.
 
     def store(self):
-        d = deferToThread(self._riak_obj.store)
-        d.addCallback(type(self))
+        d = self._manager.deferToThread(self._riak_obj.store)
+        d.addCallback(type(self), self._manager)
         return d
 
     def reload(self):
-        d = deferToThread(self._riak_obj.reload)
-        d.addCallback(type(self))
+        d = self._manager.deferToThread(self._riak_obj.reload)
+        d.addCallback(type(self), self._manager)
         return d
 
     def delete(self):
-        d = deferToThread(self._riak_obj.delete)
-        d.addCallback(type(self))
+        d = self._manager.deferToThread(self._riak_obj.delete)
+        d.addCallback(type(self), self._manager)
         return d
 
 
+def _daemon_threadpool(*args, **kw):
+    """
+    Create a threadpool that sets its threads to daemon mode so they don't
+    block process exit if they aren't properly cleaned up.
```

Comment:
The more relevant question here is whether we want to hang forever because something didn't properly close a `TxRiakManager` object.

Calling `TxRiakManager.close_manager()` will wait for the threads to finish what they're doing and stop them. If that doesn't happen, idle threads will hang around forever and if they're non-daemon threads they'll keep the process alive past the termination of the main thread.